### DISCUSSION TITLE
引用 tdesign-common 作为模块使用，保证同步更新样式库

### DISCUSSION THREAD

**Discussion by sangyuxiaowu** [AchievedOwner/TDesignBlazor#177] [讨论]
是否考虑按照 [tdesign-vue-next](https://github.com/Tencent/tdesign-vue-next) 等官方库的方式引用 [tdesign-common](https://github.com/Tencent/tdesign-common) 作为子模块。

几个月前我想实现 tdesign 的版本，看了 tdesign-vue-next 的，参照刚做了好了个button，然后帖子有回复说是有社区实现了，搜到了这个项目，然后就没再自己搞了

**Comment by teacher-zhou** [comment]
> 是否考虑按照 [tdesign-vue-next](https://github.com/Tencent/tdesign-vue-next) 等官方库的方式引用 [tdesign-common](https://github.com/Tencent/tdesign-common) 作为子模块。
> 
> 几个月前我想实现 tdesign 的版本，看了 tdesign-vue-next 的，参照刚做了好了个button，然后帖子有回复说是有社区实现了，搜到了这个项目，然后就没再自己搞了

能不能提供一个具体的思路以供参考和讨论

**Reply by sangyuxiaowu** [reply]
我看项目是全局的引用  tdesign.css 和 tdesign.min.css。可以将[tdesign-common](https://github.com/Tencent/tdesign-common) 作为git modules 引入项目，通过 BuildBundlerMinifier 来生成这两个文件，这样 tdesign-common 更新后更新子模块就可以了

**Reply by sangyuxiaowu** [reply]
我还有个想法，是不是按需引用css会更好呢？

**Reply by teacher-zhou** [reply]
> 我还有个想法，是不是按需引用css会更好呢？

如何按需呢？

**Reply by teacher-zhou** [reply]
> 我看项目是全局的引用 tdesign.css 和 tdesign.min.css。可以将[tdesign-common](https://github.com/Tencent/tdesign-common) 作为git modules 引入项目，通过 BuildBundlerMinifier 来生成这两个文件，这样 tdesign-common 更新后更新子模块就可以了

最好有示例代码以更直观的看到你的想法并进行讨论

**Reply by sangyuxiaowu** [reply]
> > 我看项目是全局的引用 tdesign.css 和 tdesign.min.css。可以将[tdesign-common](https://github.com/Tencent/tdesign-common) 作为git modules 引入项目，通过 BuildBundlerMinifier 来生成这两个文件，这样 tdesign-common 更新后更新子模块就可以了
> 
> 最好有示例代码以更直观的看到你的想法并进行讨论

https://github.com/sangyuxiaowu/TdesignBlazor  按需引用可能会跟暗黑冲突，没测试

**Reply by teacher-zhou** [reply]
为啥需要按需加载呢？我看最后编译出来也是一个完整的 css

**Reply by sangyuxiaowu** [reply]
> 为啥需要按需加载呢？我看最后编译出来也是一个完整的 css

那个不是完整的啊，只有global、reset、theme 这样公共的，组件的css都是单独的，比如Button.razor.css 这个按需不是关注点啦，可能还有问题吧，也行。主要是看子模块。按需就算了，也许没啥意义，就是可能节省一些资源，但也没啥用吧。

**Reply by wuxinheng** [reply]
按需引入非常有必要，用户更注重渲染、响应的速度，我之前在实现anchor 的时候也设想过直接引用腾讯提供的代码来实现，后面发现这个太难了并不是想象中那么容易。这需要一个前端大佬来做强有力的支撑。

**Reply by sangyuxiaowu** [reply]
是的，我会偏向于使用按需加载，大道至简

**Reply by sangyuxiaowu** [reply]
就是工作量会稍微大一点，但是这个是一劳永逸的，css样式什么的都是[tdesign-common](https://github.com/Tencent/tdesign-common) 的，写好处理脚本就完事了，官方升级，css随之升级就可以了

**Reply by teacher-zhou** [reply]
> 就是工作量会稍微大一点，但是这个是一劳永逸的，css样式什么的都是[tdesign-common](https://github.com/Tencent/tdesign-common) 的，写好处理脚本就完事了，官方升级，css随之升级就可以了

那么怎么做？在每个组件下弄一个 css？例如 `Button.razor.css`

**Reply by sangyuxiaowu** [reply]
是哦，这个原样式库就是单独的一个个的，Button.razor.css 也是配置好，自动生成的
